lib/cylc/network/server.py

```diff
@@ -0,0 +1,673 @@
+        """Prepare job file for a task.
+
+        items[0] is an identifier for matching a task proxy.
+        """
+        if not isinstance(items, list):
+            items = [items]
+        self.schd.command_queue.put(('dry_run_tasks', (items,),
+                                    {'check_syntax': check_syntax}))
+        return (True, 'Command queued')
+
+    @authorise(Priv.CONTROL)
+    @ZMQServer.expose
+    def expire_broadcast(self, cutoff=None):
+        """Clear all settings targeting cycle points earlier than cutoff."""
+        return self.schd.task_events_mgr.broadcast_mgr.expire_broadcast(cutoff)
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_broadcast(self, task_id=None):
+        """Retrieve all broadcast variables that target a given task ID."""
+        return self.schd.task_events_mgr.broadcast_mgr.get_broadcast(task_id)
+
+    @authorise(Priv.IDENTITY)
+    @ZMQServer.expose
+    def get_cylc_version(self):
+        """Return the cylc version running this suite."""
+        return CYLC_VERSION
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_graph_raw(self, start_point_string, stop_point_string,
+                      group_nodes=None, ungroup_nodes=None,
+                      ungroup_recursive=False, group_all=False,
+                      ungroup_all=False):
+        """Return raw suite graph."""
+        # Ensure that a "None" str is converted to the None value.
+        if stop_point_string is not None:
+            stop_point_string = str(stop_point_string)
+        return self.schd.info_get_graph_raw(
+            start_point_string, stop_point_string,
+            group_nodes=group_nodes,
+            ungroup_nodes=ungroup_nodes,
+            ungroup_recursive=ungroup_recursive,
+            group_all=group_all,
+            ungroup_all=ungroup_all)
+
+    @authorise(Priv.DESCRIPTION)
+    @ZMQServer.expose
+    def get_suite_info(self):
+        """Return a dict containing the suite title and description."""
+        return self.schd.info_get_suite_info()
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_suite_state_summary(self):
+        """Return the global, task, and family summary data structures."""
+        return self.schd.info_get_suite_state_summary()
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_task_info(self, names):
+        """Return info of a task."""
+        if not isinstance(names, list):
+            names = [names]
+        return self.schd.info_get_task_info(names)
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_task_jobfile_path(self, task_id):
+        """Return task job file path."""
+        return self.schd.info_get_task_jobfile_path(task_id)
+
+    @authorise(Priv.READ)
+    @ZMQServer.expose
+    def get_task_requisites(self, items=None, list_prereqs=False):
+        """Return prerequisites of a task."""
+        if not isinstance(items, list):
+            items = [items]
```

Comment:
In the old REST API, whether items is a list or not may be ambiguous. I cannot see why this will still be the case in the new API?

Reply:
Will strip these typing statements.